src/ape_accounts/_cli.py

```diff
@@ -64,22 +64,36 @@ def _list(cli_ctx, show_all_plugins):
             click.echo()
 
 
-@cli.command(short_help="Create a new keyfile account with a random private key")
+@cli.command(short_help="Create a new keyfile account with a random mnemonic seed phrase")
+@click.option(
+    "--word-count",
+    "word_count",
+    help="Number of words to use to generate seed phrase",
+    default=12,
+    show_default=True,
+)
+@click.option(
+    "--hdpath",
+    "custom_hd_path",
+    help="Specify an HD path for deriving seed phrase",
+    default="m/44'/60'/0'/0/0",
+    show_default=True,
+)
 @non_existing_alias_argument()
 @ape_cli_context()
-def generate(cli_ctx, alias):
+def generate(cli_ctx, alias, hide_mnemonic, word_count, custom_hd_path):
     path = _get_container().data_folder.joinpath(f"{alias}.json")
-    extra_entropy = click.prompt(
-        "Add extra entropy for key generation...",
-        hide_input=True,
-    )
-
-    account = EthAccount.create(extra_entropy)
     passphrase = click.prompt(
         "Create Passphrase",
         hide_input=True,
         confirmation_prompt=True,
     )
+    EthAccount.enable_unaudited_hdwallet_features()
+    account, mnemonic = EthAccount.create_with_mnemonic(
+        passphrase, word_count, "english", custom_hd_path
+    )
+    cli_ctx.logger.success(f"Newly generated mnemonic is: {mnemonic}")
```

Comment:
Add `hide_mnemonic` to options above

```suggestion
    if not hide_mnemonic and click.confirm("Show mnemonic?", default=True):
        cli_ctx.logger.success(f"Newly generated mnemonic is: {mnemonic}")
```

Reply:
ah my misunderstanding I thought we didn't need hide-mnemonic since it doesn't show in bash history and there is no other way to get the mnemonic other than at the generation point?

Reply:
So yes, it won't show in bash history, but there are some scenarios where there might be screen recording apps in use (or like ssh tunneling) where displaying the mnemonic is not a good idea. Having both the flag and the confirmation would allow use cases that are more sensitive (or don't care, e.g. using the docker container) to skip this display for extra safety

Reply:
ok great thanks!